### ISSUE TITLE
Cannot get mysql container to run

### ISSUE THREAD

**Issue by ucfchandra** [docker-library/mysql#1029]
Stuck on an error where it says 

```
[InnoDB] InnoDB initialization has started.
2024-02-19T20:20:55.218851Z 1 [ERROR] [MY-012526] [InnoDB] Upgrade is not supported after a crash or shutdown with innodb_fast_shutdown = 2. This redo log was created with MySQL 5.7.44, and it appears logically non empty. Please follow the instructions at http://dev.mysql.com/doc/refman/8.3/en/upgrading.html
2024-02-19T20:20:55.218966Z 1 [ERROR] [MY-012930] [InnoDB] Plugin initialization aborted with error Generic error.
2024-02-19T20:20:55.664309Z 1 [ERROR] [MY-010334] [Server] Failed to initialize DD Storage Engine
2024-02-19T20:20:55.664818Z 0 [ERROR] [MY-010020] [Server] Data Dictionary initialization failed.
2024-02-19T20:20:55.664909Z 0 [ERROR] [MY-010119] [Server] Aborting
2024-02-19T20:20:55.666696Z 0 [System] [MY-010910] [Server] /usr/sbin/mysqld: Shutdown complete (mysqld 8.3.0)  MySQL Community Server - GPL.
2024-02-19T20:20:55.666803Z 0 [System] [MY-015016] [Server] MySQL Server - end.
```

Can't seem to run the script to change it either. I've tried adding a docker entrypoint script that isn't running and copying in the innodb value into a sql file and trying to run it. 

Also tried to go in and run the mysql command and change the value but the container will just loop out this error indefinitely making it impossible for me to exec in. Please help

**Comment by tianon** [comment]
`2024-02-19T20:20:55.218851Z 1 [ERROR] [MY-012526] [InnoDB] Upgrade is not supported after a crash or shutdown with innodb_fast_shutdown = 2. This redo log was created with MySQL 5.7.44, and it appears logically non empty. Please follow the instructions at http://dev.mysql.com/doc/refman/8.3/en/upgrading.html`

Did you update from MySQL 5.7.44 to MySQL 8.3 on purpose?

**Comment by ucfchandra** [comment]
On my machine yes but not in the container. I don't understand why the inside of the container is returning this unless they did that when building the 8.3 version of mysql

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by tianon** [comment]
This line in your logs means the container is running 8.3+ now, but _was_ running 5.7.44 previously, so you need to fix that (maybe start back up with 5.7, let it repair the unclean shutdown, _then_ do a clean shutdown, _then_ let MySQL upgrade to 8+ following the upgrade doc).
